controllers/mongodb_controller.go

```diff
@@ -290,6 +292,31 @@ func (r *MongoDBReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error)
 		memoryLimit = instance.Spec.Resources.Limits.Memory().String()
 	}
 
+	// Default values
+	PVCSizeRequest := defaultPVCSize
+
+	// If PVC already exist and the value does not match the PVCSizeRequest then log information that it cannot be changed.
+	pvc := &corev1.PersistentVolumeClaim{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: "mongodbdir-icp-mongodb-0", Namespace: instance.Namespace}, pvc)
+	if err == nil {
+		PVCSizeRequest = pvc.Spec.Resources.Requests.Storage().String()
+		if instance.Spec.PVC.Resources.Requests.Storage().String() != "0" {
```

Comment:
@kgcarr I don't understand why we compare the string to "0".  Should it be ""?

Reply:
Storage is resource.Quantity, when not filled in, it's default value is 0. I originally had the same thought that it would be "", but my tests failed and I realized I need zero. The same was true for the CPU and Memory request and limit checks

Reply:
Oh, Thanks @kgcarr. Good to know that